lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -93,8 +93,11 @@ def parse(document, options = {})
           attributes['id'] = primary_data['id'] if primary_data['id']
           relationships = primary_data['relationships'] || {}
 
-          filter_fields(attributes, options)
-          filter_fields(relationships, options)
+          attributes = transform_keys(attributes, options)
+          relationships = transform_keys(relationships, options)
```

Comment:
yeah, which it is by default.
transform_keys has ` options[:key_transform] || :underscore`

So, I guess if options[key_transform] is set, maybe I should be calling transform_keys on the only/except data as well